### PR TITLE
touchstone+timestamp

### PR DESCRIPTION
**Pull request by Ayesha279**
### Description
New function is added to touchstone that get retrieve timestamp data. It is also capable to providing more then 10000 records of data
### Fixes
new code add to retrieve data with out using aggregations 
inserted a scan() function to surpass the limit of getting a 10000 data points for data collection of timestamps

### CODE REVIEW

File: src/touchstone/databases/elasticsearch.py [hunk 1]
```diff
@@ -163,3 +168,32 @@ def access_dotted_field(self, input_dict, fields):
...
+        for key, value in filters.items():
+            s = s.filter("term", **{key: value})
+        logger.debug("Finished adding filters")
+        logger.debug("Built the following query: {}".format(json.dumps(s.to_dict(), indent=4)))
+
+        output_list = []
+        for hit in s.scan():
+            output_list.append(hit.__dict__["_d_"])
+        results_df = pd.DataFrame((d.to_dict() for d in s.scan()))
+        print(results_df)
```
Comment (rsevilla87) [inline src/touchstone/databases/elasticsearch.py:197]: Remove this print statement

Comment (rsevilla87) [inline src/touchstone/databases/elasticsearch.py:197]: We don't need these lines right?

same for `import pandas as pd` and `import elasticsearch.helpers`

File: src/touchstone/databases/elasticsearch.py [hunk 2]
```diff
@@ -163,3 +168,32 @@ def access_dotted_field(self, input_dict, fields):
             else:
                 return None
         return tmp_dict
+
+    # def gen_result_list(self, response):
+    #    output_list = []
+    #    for hit in response:
+    #        output_list.append(hit.__dict__["_d_"])
+    #    return output_list
```
Comment (chentex) [inline src/touchstone/databases/elasticsearch.py:176]: I think this can go away.

File: src/touchstone/databases/elasticsearch.py [hunk 3]
```diff
@@ -163,3 +168,32 @@ def access_dotted_field(self, input_dict, fields):
...
+        # Apply filters
+        for key, value in filters.items():
+            s = s.filter("term", **{key: value})
+        logger.debug("Finished adding filters")
+        logger.debug("Built the following query: {}".format(json.dumps(s.to_dict(), indent=4)))
+
+        output_list = []
+        for hit in s.scan():
+            output_list.append(hit.__dict__["_d_"])
+        results_df = pd.DataFrame((d.to_dict() for d in s.scan()))
```
Comment (chentex) [inline src/touchstone/databases/elasticsearch.py:196]: I see something weird here, why are you scanning two times ES? can't we use the same response we already got?

File: src/touchstone/databases/elasticsearch.py (hunk 1 above)
Comment (chentex) [inline src/touchstone/databases/elasticsearch.py:197]: i think this can also be deleted

**Review by chentex — CHANGES_REQUESTED** [review]

File: config/ocm-requests.json [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+    "ocm-requests": [
+      {
+        "filter": {
+          "test_name.keyword": "self-terms-review"
+        },
+        "buckets":[
+          "code",
+          "latancy_nc"
+        ],
+                "timeseries": true
```
Comment (dry923) [inline config/ocm-requests.json:12]: Fix alignment to be under buckets

File: src/touchstone/databases/elasticsearch.py [hunk 5]
```diff
@@ -1,8 +1,13 @@
 import logging
+
+# include pandas
```
Comment (dry923) [inline src/touchstone/databases/elasticsearch.py:3]: Remove if not needed.

File: src/touchstone/databases/elasticsearch.py [hunk 6]
```diff
@@ -163,3 +168,32 @@ def access_dotted_field(self, input_dict, fields):
...
+
+    # def gen_result_list(self, response):
+    #    output_list = []
+    #    for hit in response:
+    #        output_list.append(hit.__dict__["_d_"])
+    #    return output_list
+
+    def get_timeseries_results(self, uuid, compute_map, index, identifier):
+
+        # not aggreated data
```
Comment (dry923) [inline src/touchstone/databases/elasticsearch.py:180]: Random comment or actual comment?

File: src/touchstone/compare.py [hunk 7]
```diff
@@ -169,17 +164,37 @@ def main(args):
...
-                mergedicts(result, index_json)
-
-                compute_header = extract_headers(compute, args.identifier)
+                if "aggregations" in compute:
+                    result = database_instance.emit_compute_dict(
+                        uuid=uuid, compute_map=compute, index=index, identifier=args.identifier,
+                    )
+                    mergedicts(result, main_json)
+                    mergedicts(result, index_json)
+                    compute_header = []
```
[the remark this answers is not in the document]
Comment (Ayesha279) [reply src/touchstone/compare.py:177]: the extract_headers was used cause we had a bunch of aggregation, right now as the file is pretty simple and in not a bunch of nested dictionaries we don't really need it.

File: src/touchstone/databases/elasticsearch.py (hunk 6 above)
Comment (Ayesha279) [reply src/touchstone/databases/elasticsearch.py:170]: actual comment... the start of code for non-aggregated data (will change the spelling tho)

File: src/touchstone/compare.py (hunk 7 above)
[the remark this answers is not in the document]
Comment (chentex) [reply src/touchstone/compare.py:177]: For what I see is that the method and this code do the exact same thing, maybe I am missing something, but we should keep the function.
https://github.com/cloud-bulldozer/benchmark-comparison/blob/0c6521c897d2217f72fbefa83f68cd2a632055f3/src/touchstone/utils/lib.py#L39

File: config/ocm-requests.json [hunk 8]
```diff
@@ -0,0 +1,16 @@
+{
+  "elasticsearch": {
+    "ocm-requests": [
+      {
+        "filter": {
+          "test_name.keyword": "self-terms-review"
+        },
+        "buckets":[
+          "code",
+          "latancy_nc"
```
Comment (chentex) [inline config/ocm-requests.json:10]: should be `latency_ns`

**Review by chentex — APPROVED** [review]
Thank for the patience, LGTM.

@dry923 @rsevilla87 can you take a final look before we squash?

File: src/touchstone/compare.py [hunk 9]
```diff
@@ -171,21 +166,52 @@ def main(args):
...
                     )
-                mergedicts(result, main_json)
-                mergedicts(result, index_json)
-
-                compute_header = extract_headers(compute, args.identifier)
-
+                    if not result:
+                        logger.error(
+                            f"Error: Issue capturing results from {args.database} using config {compute}"
+                        )
```
Comment (jtaleric) [inline src/touchstone/compare.py:180]: return after empty result.

File: src/touchstone/compare.py [hunk 10]
```diff
@@ -204,6 +230,7 @@ def main(args):
         output_file.write(json.dumps(main_json, indent=4))
     elif args.output == "yaml":
         output_file.write(yaml.dump(main_json, allow_unicode=True))
+    logger.debug("Script ends here")
```
Comment (jtaleric) [inline src/touchstone/compare.py:233]: remove debug code

File: src/touchstone/compare.py [hunk 11]
```diff
@@ -171,21 +166,54 @@ def main(args):
             index_json = {}
             # Iterate through UUIDs
             for uuid_index, uuid in enumerate(args.uuid):
+
```
Comment (dry923) [inline src/touchstone/compare.py:169]: extremely minor nit but unneeded empty line

File: src/touchstone/databases/elasticsearch.py [hunk 12]
```diff
@@ -163,3 +164,23 @@ def access_dotted_field(self, input_dict, fields):
             else:
                 return None
         return tmp_dict
+
+    def get_timeseries_results(self, uuid, compute_map, index, identifier):
+
```
Comment (dry923) [inline src/touchstone/databases/elasticsearch.py:169]: same extremely minor nit with leading empty line

**Review by dry923 — APPROVED** [review]
LGTM!

File: config/ocm-requests.json [hunk 13]
```diff
@@ -0,0 +1,16 @@
+{
+  "elasticsearch": {
+    "ocm-requests": [
+      {
+        "filter": {
+          "test_name.keyword": "self-terms-review"
+        },
+        "buckets":[
+          "code",
+          "latancy_ns"
```
Comment (chentex) [inline config/ocm-requests.json:10]: typo here `latency_ns`

**Review by chentex — CHANGES_REQUESTED** [review]
Just a small typo

**Review by chentex — APPROVED** [review]
lgtm

File: config/ocm-requests.json [hunk 14]
```diff
@@ -0,0 +1,16 @@
...
+  "elasticsearch": {
+    "ocm-requests": [
+      {
+        "filter": {
+          "test_name.keyword": "self-terms-review"
+        },
+        "buckets":[
+          "code",
+          "latency_ns"
+        ],
```
Comment (rsevilla87) [inline config/ocm-requests.json:11]: This kind of query, "timeseries", does not use buckets. Remove them from here, pls

**Review by rsevilla87 — CHANGES_REQUESTED** [review]
Hey @Ayesha279, mind you update the readme to describe this new feature?.

**Comment by chentex** [comment]
Hey @rsevilla87 I have added the changes you requested.

**Review by jtaleric — APPROVED** [review]
LGTM!

**Pull request merged**